### DISCUSSION TITLE
How insert some text in the doc before the generated Nuxt content

### DISCUSSION THREAD

**Discussion by Schetnan** [nuxt/nuxt#16695] [Questions]
We have a corporate proxy that expects that all applications have a snippet like the following at the very top of the page - before anything else.  It is used to pass information to the proxy to determine a corporate template for wrapping up a header/footer etc.  Sometimes, it needs to be dynamically updated.  In our nuxt2 apps, we just insert it at the top like the following in the app.html file.  Can anyone lead me to a solution of how I might do this in nuxt3?

```
<!--
  META_TAGS={"robots": "noindex,nofollow"}
  PAGE_TITLE=My App Title
  SITE=MySite
	TEMPLATE_NAME=auth
	ENABLE_DYNAMIC_HEAD_TAG=true
	ENABLE_DYNAMIC_TITLE=true
-->
<meta name="viewport" content="width=device-width, minimum-scale=1.0" />

<head {{ HEAD_ATTRS }}>
  {{ HEAD }}
</head>

<div {{ HTML_ATTRS }}>
  <div {{ BODY_ATTRS }}>{{ APP }}</div>
</div>
```

**Comment by TheAlexLichter** [comment]
You could use a Nitro plugin and hook into `render:response`. Ideally, `render:html` would give you the opportunity with a variable like `htmlPrepend` but sadly this isn't the case right now.

See: https://github.com/nuxt/framework/pull/7782/ https://github.com/nuxt/framework/issues/5565

**Reply by Schetnan** [reply]
@manniL Thank you for the suggestion.  I tried the Nitro plugin - render:response method.  I was able to achieve what I needed to with that method.  Thank you!

**Reply by TheAlexLichter** [reply]
Glad to hear! ☺️ Feel free to mark the post as answer then 😋
